src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -485,7 +486,7 @@ private int getNoOfFailedMonitorMetricGroup() {
 	 */
 	private void populateQSYSMonitoringMetrics(Map<String, String> stats) {
 		Objects.requireNonNull(stats);
-		if (getLogin() != null && getPassword() != null) {
+		if (!StringUtils.isNullOrEmpty(getPassword()) && !StringUtils.isNullOrEmpty(getLogin())) {
```

Comment:
please add one more UT for this case. Thanks!

Reply:
I added UT to cover this case.